src/index.ts

```diff
@@ -117,6 +117,9 @@ function isBuffer(p: XtpTyped): boolean {
 function isObject(p: XtpTyped): boolean {
   return p?.xtpType?.kind === "object"
 }
+function isUntypedObject(p: XtpTyped): boolean {
```

Comment:
I'm kind of on the fence about the name now. Perhaps we should just call it `JsonObject` and `kind=jsonobject`. I don't like that the JSON is leaking through here, but also i'm failing to see what type this would be in some languages if it wasn't json. e.g. what would this generated type look like in protobuf? It's inherently very tied to json. And maybe it's just a quirk we support but don't encourage. What are your thoughts @G4Vi ?

Reply:
maybe we even, once we go to 1.0, remove this feature and add a supported `type: jsonobject` or something.

Reply:
I'd be okay renaming it to `JsonObject` as I think json / javascript were first to adding a type like it. I'm not sure what other serialization methods support besides JSON, but it seems to have a language native type equivalent in many languages so we should keep supporting it: go (`interface`), python `dict`, perl `%hash`, etc. We should add a note that their less efficient in typed languages and discourage their use outside of prototyping.

Reply:
Perhaps we call it "FreeFormObject" object as they do in openapi: 

https://swagger.io/docs/specification/v3_0/data-models/data-types/#objects
https://swagger.io/docs/specification/v3_0/data-models/dictionaries/

When we add support for `additionalProperties: true` https://github.com/dylibso/xtp/issues/949 we can re-use this intermediate type as well.